rest/tests/base.py

```diff
@@ -0,0 +1,389 @@
+            return as_json(resp)
+
+    def create_repository(self, repo_dict=DEFAULT_REPO_DICT,
+                          expected_status=HTTP_201_CREATED,
+                          skip_assert=False):
+        """Helper function to create repository"""
+
+        resp = self.client.post(REPO_BASE, repo_dict)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_201_CREATED:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+
+            self.assertIn(reverse(
+                'repository-detail', kwargs={
+                    'repo_slug': result_dict['slug'],
+                }
+            ), resp['Location'])
+
+            return result_dict
+
+    def patch_repository(self, repo_slug, repo_dict,
+                         expected_status=HTTP_200_OK, skip_assert=False):
+        """Update a repository"""
+        resp = self.client.patch(
+            '{repo_base}{repo_slug}/'.format(
+                repo_slug=repo_slug,
+                repo_base=REPO_BASE,
+            ),
+            json.dumps(repo_dict),
+            content_type='application/json',
+        )
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+            return result_dict
+
+    def put_repository(self, repo_slug, repo_dict,
+                       expected_status=HTTP_200_OK, skip_assert=False):
+        """Replace a repository"""
+        resp = self.client.put(
+            '{repo_base}{repo_slug}/'.format(
+                repo_slug=repo_slug,
+                repo_base=REPO_BASE,
+            ),
+            json.dumps(repo_dict),
+            content_type='application/json',
+        )
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+
+            return result_dict
+
+    def get_repository(self, repo_slug, expected_status=HTTP_200_OK,
+                       skip_options_head_test=False):
+        """Get a repository"""
+        url = '{repo_base}{slug}/'.format(
+            slug=repo_slug,
+            repo_base=REPO_BASE,
+        )
+        if not skip_options_head_test:
+            self.assert_options_head(url, expected_status=expected_status)
+
+        resp = self.client.get(url)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            return as_json(resp)
+
+    def delete_repository(self, repo_slug,
```

Comment:
This is tested, but it's complicated by the fact that deletes aren't currently allowed at all for repositories (see #191). TestRest.test_repositories currently tries the delete and verifies that the count hasn't changed since the delete failed